src/vs/workbench/contrib/debug/common/loadedScriptsPicker.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as nls from 'vs/nls';
+import { matchesFuzzy } from 'vs/base/common/filters';
+import { Source } from 'vs/workbench/contrib/debug/common/debugSource';
+import { IQuickInputService, IQuickPickItem, IQuickPickSeparator } from 'vs/platform/quickinput/common/quickInput';
+import { IDebugSession } from 'vs/workbench/contrib/debug/common/debug';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { getIconClasses } from 'vs/editor/common/services/getIconClasses';
+import { IModelService } from 'vs/editor/common/services/model';
+import { ILanguageService } from 'vs/editor/common/languages/language';
+
+export interface IPickerLoadedScriptItem extends IQuickPickItem {
+	accept(): void;
+}
+
+
+/**
+ * This function takes a regular quickpick and makes one for loaded scripts that has persistent headers
+ * e.g. when some picks are filtered out, the ones that are visible still have its header.
+ */
+export async function showLoadedScriptMenu(sessions: IDebugSession[], quickInputService: IQuickInputService, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService) {
+
+	const quickPick = quickInputService.createQuickPick<IPickerLoadedScriptItem>();
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "Search loaded scripts by name");
+	quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	const changeListener = quickPick.onDidChangeValue(async () => {
+		quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	});
+	const acceptListener = quickPick.onDidAccept(() => {
+		const selectedItem = quickPick.selectedItems[0];
+		selectedItem.accept();
+		acceptListener.dispose();
```

Comment:
Try creating a DisposableStore and adding the listeners and quickpick to it at the moment they are created. I like this better because then you only have one thing to dispose, and it's also more obvious that every disposable is tracked somewhere.